File: ruby/fizzBuzz.rb
def fizzBuzz( x, y, z )
	
	result = Array.new
	
	x = x.to_i
	y = y.to_i
	z = z.to_i
	
	(1..z).each do |i|
		
		# Divisible by both.
		if i % x == 0 and i % y == 0
			result.push("FB")
		elsif i % x == 0
			result.push("F")
		elsif i % y == 0
			result.push("B")
		else
			result.push(i)
		end
			
	end
	
	return result.join(" ")
end

output = Array.new

File.open(ARGV[0]).each_line do |line|
# Do something with line, ignore empty lines
#...
	print line
	if line == ""
		next
	end
	
	output.push(fizzBuzz(*line.split(" ") ))
	
end

print output.join("\n")

File: js/FizzBuzz.js

var fs = require('fs');
var output = [];

// FizzBuzz!
function fizzBuzz( x, y, z ) {
    'use strict';

    var result = [];

    for ( var i = 1; i <= z; i++ ) {

        // Divisible by both.
        if( ( i % x ) === 0 && (i % y) === 0 )
            result.push( 'FB' );

        // Divisible by X
        else if( ( i % x ) === 0 )
            result.push( 'F' );

        // Divisible by Y
        else if( ( i % y ) === 0 )
            result.push( 'B' );

        // Not divisible.
        else
            result.push( i );
    }

    return result.join(' ');
}

fs.readFileSync( process.argv[2] )
    .toString()
    .split( '\n' )
    .forEach( function( line ) {
        if ( line == '' ) return;

        output.push( fizzBuzz.apply( null, line.split(' ') ) );
    });

console.log( output.join( '\n' ) );
